### PR TITLE
Rails Capstone Budget-App

### PR DESCRIPTION
**Pull request by ekenecf**
Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the signup and log-in pages.
Sign up and login pages

The user should be able to register in the app with their full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (categories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page, the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user clicks (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: app/views/deals/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+       <%= form.text_field :name, class:"form-control mb-3" %>
+
+       <%= form.label :amount %>
+       <%= form.text_field :amount, class:"form-control mb-3" %>
+
+       <%= form.select :group_id, @groups&.map{ |group| [group.name, group.id] }, { :prompt => 'Select Group'}, class:"form-control mb-3" %>
+
+       <%= form.submit "Create Deal", class:"btn btn-primary"%>
+    </div>
+<% end %>
```
Comment (medaminedev66) [inline app/views/deals/new.html.erb:16]: - [ ] Following the project requirements, on "Add a new transaction" page, the user should be able to select more than one category. Remember that the relation between category and transaction is a many-to-many relationship.

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <% if @groups.empty? %>
+    <p>No group added yet!</p>
+  <% end %>
+
+<% @groups.each do |group| %>
+    <p>Name of group: <%= group.name %>
+    <%= button_to "Delete group", user_group_path(current_user, group), method: :delete %>
+    <p><%= link_to "Group detail", user_group_path(current_user, group) %></p>
+<% end %>
+    <p><%= link_to "Back to user", users_path %></p>
```
Comment (medaminedev66) [inline app/views/groups/index.html.erb:11]: - [ ] Following the project requirements, for each category, the user should see the total amount of all the transactions that belong to that category.

File: app/views/deals/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<%= form_with model: @deal, url: user_deals_path, method: :post do |form| %>
+
+    <p><%= link_to "Back to Deals page", user_deals_path, class:"btn btn-primary" %></p>
+
```
Comment (medaminedev66) [inline app/views/deals/new.html.erb:4]: - [ ] "Add a new transaction" page should have a button that takes to the transactions page. The button that you added takes the user to the `all_deals` page, not the deal page.

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<h1> All groups </h1>
```
Comment (medaminedev66) [inline app/views/groups/index.html.erb:1]: - [ ] In the mobile version of your app, the hamburger button does not work. I recommend fixing that so the user is able to navigate between pages

File: app/views/deals/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<h1 id= "deals_header"> All deals </h1>
+<div class="jumbotron">
+  <div class="container text-center">
+    <p>Total_transaction: $<%= @total_deal %></p>
+
+        <% @deals.each do |deal| %>
+            <p>Deal name: <%= deal.name %> </p>
+            <p>Deal amount: $<%= deal.amount %> </p> <br>
+        <%end %>
```
Comment (medaminedev66) [inline app/views/deals/index.html.erb:9]: - [ ] For a given category, the list of transactions should be ordered by the most recent as per the project requirements.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ekenecf  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional
- There are no linter errors
- You have a descriptive PR
- Your web app runs without errors
- The project is deployed properly, and the home and detail page works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/deals/new.html.erb (hunk 1 above)
Comment (ekenecf) [reply app/views/deals/new.html.erb:16]: Hi @medaminedev66, On the "Add a new transaction" page, the user can select more than one category as stated in the requirement

File: app/views/deals/index.html.erb (hunk 5 above)
Comment (ekenecf) [reply app/views/deals/index.html.erb:9]: @medaminedev66 You can check my deals controller to see that I have ordered the deals by the most recent

File: app/views/deals/new.html.erb (hunk 1 above)
Comment (medaminedev66) [reply app/views/deals/new.html.erb:16]: Yes 😃

File: app/controllers/groups_controller.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  def index
+    @groups = Group.includes(:deals).where(user: current_user)
+    uder_deal = current_user.deals.sum(&:amount)
+    puts uder_deal
+  end
+
+  def show
+    @deals = @group.deals
+    @total_expenses = @deals.includes(:groups).where(user: current_user).sum(:amount)
+  end
```
Comment (medaminedev66) [inline app/controllers/groups_controller.rb:13]: Hi @ekenecf. As I can see, this is the controller of the transactions page, and the deals are not ordered by the most recent.

image

**Review by Davidosky007 — APPROVED** [review]
Hi @ekenecf 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**